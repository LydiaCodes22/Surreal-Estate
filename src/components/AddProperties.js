import React from "react";

const AddProperties = () => (
  <div className="add-properties">
    <h1>Add Properties</h1>
    <form>
      <label>
        Title:
        <input type="text" name="title" />
      </label>
    </form>{" "}
  </div>
);

export default AddProperties;
