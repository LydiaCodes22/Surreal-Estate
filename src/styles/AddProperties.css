.add-properties {
    border-radius: 5px;
    background-color: rgb(242, 240, 239);
    padding: 20px;
}
input, select {
    width: 600px;
    max-width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
  }