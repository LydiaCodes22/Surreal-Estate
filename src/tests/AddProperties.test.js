import { render, screen } from "@testing-library/react";
import React from "react";
import AddProperties from "../components/AddProperties";

describe("the add properties component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<AddProperties />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<AddProperties />);
    expect(screen.getByRole("heading", { name: /add/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /add/i })).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/title/i)).toBeInTheDocument();
    expect(screen.getByText(/type/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/bedrooms/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/bathrooms/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/price/i)).toBeInTheDocument();
    expect(screen.getByText(/location/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument();
  });
});
