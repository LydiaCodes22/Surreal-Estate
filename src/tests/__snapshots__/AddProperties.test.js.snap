// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the add properties component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="add-properties"
  >
    <h1>
      Add Properties
    </h1>
    <form
      class="add-properties-form"
    >
      <label
        class="form-label"
        for="title"
      >
        Property title
      </label>
      <input
        class="form-input"
        id="title"
        name="title"
        placeholder="Property title"
        required=""
        type="text"
        value=""
      />
      <label
        class="form-label"
        for="type"
      >
        Property type
      </label>
      <select
        class="form-input"
        id="type"
        name="type"
      >
        <option
          value="detached"
        >
          Detached
        </option>
        <option
          value="semi-detached"
        >
          Semi-detached
        </option>
        <option
          value="terrace"
        >
          terrace
        </option>
        <option
          value="end of terrace"
        >
          End of terrace
        </option>
        <option
          value="cottage"
        >
          Cottage
        </option>
        <option
          value="bungalow"
        >
          Bungalow
        </option>
      </select>
      <label
        class="form-label"
        for="bedrooms"
      >
        Bedrooms
      </label>
      <input
        class="form-input"
        id="bedrooms"
        name="bedrooms"
        placeholder="Number of bedrooms"
        required=""
        type="number"
        value=""
      />
      <label
        class="form-label"
        for="bathrooms"
      >
        Bathrooms
      </label>
      <input
        class="form-input"
        id="bathrooms"
        name="bathrooms"
        placeholder="Number of bathrooms"
        required=""
        type="number"
        value=""
      />
      <label
        class="form-label"
        for="price"
      >
        Â£
      </label>
      <input
        class="form-input"
        id="price"
        min="0.01"
        name="price"
        placeholder="Price"
        step="0.01"
        type="number"
        value=""
      />
      <label
        class="form-label"
        for="city"
      >
        Location:
      </label>
      <select
        class="form-input"
        id="city"
        name="city"
      >
        <option
          value="Manchester"
        >
          Manchester
        </option>
        <option
          value="Leeds"
        >
          Leeds
        </option>
        <option
          value="Sheffield"
        >
          Sheffield
        </option>
        <option
          value="Liverpool"
        >
          Liverpool
        </option>
      </select>
      <label
        class="form-label"
        for="email"
      >
        Email 
      </label>
      <input
        class="form-input"
        id="email"
        name="email"
        placeholder="Contact email"
        required=""
        type="email"
        value=""
      />
      <button
        class="form-input"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
